filament/backend/src/vulkan/memory/ResourcePointer.h

```diff
@@ -59,6 +60,11 @@ struct resource_ptr {
     static enabled_resource_ptr<B> cast(ResourceManager* resManager,
             Handle<B> const& handle) noexcept {
         D* ptr = resManager->handle_cast<D*, B>(handle);
+        if (UTILS_UNLIKELY(ptr->isHandleConsideredDestroyed())) {
+            FILAMENT_CHECK_PRECONDITION(!ptr->isHandleConsideredDestroyed())
+                    << "Handle id=" << ptr->id << " (" << getTypeStr(ptr->restype)
+                    << ") is being used after it has been freed";
+        }
```

Comment:
you don't need the `if` here, `FILAMENT_CHECK_PRECONDITION` already does all the `likely/unlikely` shenanigans

Reply:
I was trying to avoid hitting the switch in `getTypeStr` unless necessary, hence the roundabout logic here.
Do you think it's worth it or not? (or maybe it just going to be speculatively executed anyways?).

Reply:
Actually, `FILAMENT_CHECK_PRECONDITION` is pretty sick. If the condition is false, it *will not* execute at all what comes after the `<<`. I know it's magic.

Reply:
done